src/modules/trip-main/view/trip-main-info-view.js

```diff
@@ -1,21 +1,114 @@
 import AbstractView from '../../../framework/view/abstract-view.js';
+import { DatetimeFormat, convertDatetime, isSameMonth } from '../../../global/utils/date.js';
 
-const createTripMainInfoTemplate = () => (/*html*/`
-    <section class="trip-main__trip-info  trip-info">
-      <div class="trip-info__main">
-        <h1 class="trip-info__title">Amsterdam &mdash; Chamonix &mdash; Geneva</h1>
-        <p class="trip-info__dates">Mar 18&nbsp;&mdash;&nbsp;20</p>
-      </div>
+//! Определение наименования маршрута
+//! ------------------------------------------------------
 
-      <p class="trip-info__cost">
-        Total: &euro;&nbsp;<span class="trip-info__cost-value">1230</span>
-      </p>
-    </section>`
-);
+const getDestinationItem = (destinations, eventItem) => destinations.find((item) => item.id === eventItem.destination);
+
+const getTitle = ({ destinations, events }) => {
+  const destinationNamesList = events
+    .map((event) => getDestinationItem(destinations, event))
+    .map((item) => item.name);
+
+  if (destinationNamesList.length <= 3) {
+    return destinationNamesList.join('&nbsp;&mdash;&nbsp;');
+  }
+
+  return `${destinationNamesList[0]}&nbsp;&mdash;&nbsp;...&nbsp;&mdash;&nbsp;${destinationNamesList.at(-1)}`;
+};
+
+
+//! Определение дат начала и конца путешествия
+//! ------------------------------------------------------
+
+const sortByDateFrom = (first, second) => {
+  if (first.dateFrom > second.dateFrom) {
+    return 1;
+  }
+  if (first.dateFrom < second.dateFrom) {
+    return -1;
+  }
+};
+
+const sortByDateTo = (first, second) => {
+  if (first.dateTo > second.dateTo) {
+    return -1;
+  }
+  if (first.dateTo < second.dateTo) {
+    return 1;
+  }
+};
+
+const getTripDates = ({ events }) => {
+  if (!events.length) {
+    return '';
+  }
+
+  const ascSortedEventsByDateFrom = events.slice().sort(sortByDateFrom);
+  const descSortedEventsByDateTo = events.slice().sort(sortByDateTo);
```

Comment:
Можно деструктуризацией сразу достать ивент чтобы не брать [0]
```suggestion
  const [lastEvent] = events.slice().sort(sortByDateTo);
```